tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts

```diff
@@ -441,46 +435,37 @@ export class MeshViewer extends THREE.EventDispatcher {
         metalness: 0,
       },
     }) as any;
-    let geometry = new THREE.Geometry();
-    vertices.forEach(function (point) {
-      let p = new THREE.Vector3(point[0], point[1], point[2]);
-      const scale = 1;
-      p.x = point[0] * scale;
-      p.y = point[1] * scale;
-      p.z = point[2] * scale;
-      geometry.vertices.push(p);
-    });
-    faces.forEach(function (face_indices) {
-      let face = new THREE.Face3(
-        face_indices[0],
-        face_indices[1],
-        face_indices[2]
-      );
-      if (colors && colors.length) {
-        const face_colors = [
-          colors[face_indices[0]],
-          colors[face_indices[1]],
-          colors[face_indices[2]],
-        ];
-        for (let i = 0; i < face_colors.length; i++) {
-          const vertex_color = face_colors[i];
-          let color = new THREE.Color(
-            vertex_color[0] / 255,
-            vertex_color[1] / 255,
-            vertex_color[2] / 255
-          );
-          face.vertexColors.push(color);
-        }
-      }
-      geometry.faces.push(face);
-    });
+    const geometry = new THREE.BufferGeometry();
+    // Flattens the [N, 3] shaped `vertices` into a N*3 length array.
+    const flatPoints = new Float32Array(vertices.flat());
+    geometry.setAttribute('position', new THREE.BufferAttribute(flatPoints, 3));
+
+    // Flattens the [N, 3] shaped `faces` into a N*3 length array of indices.
+    const vertexIndicesPerFace = new Uint16Array(faces.flat());
```

Comment:
Why "face" is an integer? (while points are float) What is the meaning of it?

Reply:
Updated the comment.

This array contains indices which correspond to "items" in the 'positions' attribute.  Added an example in the comment to illustrate how each "item" is a triplet of values (XYZ).

Most of my understanding was gained through reading this helpful site, and debugging
https://threejsfundamentals.org/threejs/lessons/threejs-custom-buffergeometry.html